### PR TITLE
Add ComparisonTrait::isSameMonth()/ComparisonTrait::isSameYear()

### PR DESCRIPTION
**Pull request by andrii-pukhalevych**
PR for #382

### CODE REVIEW

File: src/ChronosInterface.php [hunk 1]
```diff
@@ -534,6 +534,22 @@ public function isLeapYear(): bool;
      */
     public function isSameDay(ChronosInterface $dt): bool;
 
+    /**
+     * Checks if the passed in date is the same month as the instance current month.
+     *
+     * @param \Cake\Chronos\ChronosInterface $dt The instance to check against.
+     * @return bool
+     */
+    public function isSameMonth(ChronosInterface $dt): bool;
```
Comment (ADmad) [inline src/ChronosInterface.php:543]: Interfaces can only we changed in major releases, so this needs to target 3.x.

Comment (andrii-pukhalevych) [reply src/ChronosInterface.php:543]: Can it be added to 2.x without change of interface but only implementation?

Comment (ADmad) [reply src/ChronosInterface.php:543]: Theoretically it can be but IMO this isn't an important enough addition to deviate from the interface.

Comment (andrii-pukhalevych) [reply src/ChronosInterface.php:543]: Ok, but can you help with rebase? For me it gives conflicts in files. Or should create new PR?

Comment (ADmad) [reply src/ChronosInterface.php:543]: Creating a new PR would be easier :)

**Pull request closed without merging**